More updates from Patch 1.0.0.290

tyrken opened the pull request:
Here is another attempt at helping with the massive changes, starting with just a couple of files in case I'm doing something wrong - please provide feedback as it's very difficult doing it blind - no syntax/lint & little lua knowledge.  Thankfully found a old but useful GUI for 5-way merges in http://diffuse.sourceforge.net/index.html I drive with a space-normalisation script.

Sometimes it's been easier to start from a tab-to-spaced form of the updated 290 code and just add back in the CQUI changes that, but sorry this makes a lot of diffs to your CQUI expansion2 branch.

* popups/techciviccompletedpopup.lua
* popups/mappinlistpanel.xml (no changes)
* popups/wonderbuiltpopup.lua
* screens/civicstree.lua
* Screens/governmentscreen.lua
* Screens/governmentscreen.xml
* Screens/techtree.lua
* ToolTips/plottooltip.lua
* strategicview_mapplacement.lua
* techandcivicsupport.lua
* toppanel.lua
* toppanel.xml
* unitflagmanager.lua
* unitflagmanager.xml
* worldinput.lua
* worldtracker.lua
* worldtracker.xml
* worldview/plotinfo.lua
* worldview/plotinfo.xml (no changes needed)
* worldviewiconsmanager.lua

Many apologies @accatyyc I didn't notice until completed that you'd also started at the bottom working on worldviewiconsmanager.lua, please review my changes against what you have!

joelekstrom commented:
No problem, I'm not really sure what I'm doing either - but @Azurency posted a helpful review on my PR, so check that out as well.

ihendriks commented on Assets/UI/WorldView/plotinfo.lua:
```diff
@@ -878,24 +1003,98 @@ function OnClearDistrictPlacementShadowHexes()
+  elseif layerNum == m_PurchasePlot then
     ShowPurchases();
     RealizeShadowMask();
-    --RealizeTilt();
```
Same as above: This is a CQUI change, should probably leave this commented out.

tyrken replied:
Thanks, missed that, no commenting out there & similar

ihendriks commented on Assets/UI/WorldView/plotinfo.lua:
```diff
@@ -878,24 +1003,98 @@ function OnClearDistrictPlacementShadowHexes()
+  if layerNum == m_CitizenManagement then
     ShowCitizens();
     RealizeShadowMask();
-    --RealizeTilt();
```
This is a CQUI change, should probably leave this commented out.

ihendriks commented on Assets/UI/WorldView/plotinfo.lua:
```diff
@@ -906,21 +1105,40 @@ end
+  if  layerNum == m_CitizenManagement then
     HideCitizens();
     RealizeShadowMask();
-    --RealizeTilt();
```
Same as above: This is a CQUI change, should probably leave this commented out.

ihendriks commented on Assets/UI/WorldView/plotinfo.lua:
```diff
@@ -906,21 +1105,40 @@ end
+  elseif  layerNum == m_PurchasePlot then
     HidePurchases();
     RealizeShadowMask();
-    --RealizeTilt();
```
Same as above: This is a CQUI change, should probably leave this commented out.

ihendriks commented on Assets/UI/WorldView/plotinfo.lua:
```diff
@@ -906,21 +1105,40 @@ end
+  elseif layerNum == m_CityDetails then
+    HideCityDetails();
+    RealizeShadowMask();
+    RealizeTilt();
```
I'm gonna go ahead and assume we'll want this one commented out as well, as we commented out all the others before.

tyrken replied:
Agreed

ihendriks commented on Assets/UI/WorldView/plotinfo.lua:
```diff
@@ -937,15 +1155,6 @@ function OnPlayerTurnActivated( ePlayer:number, isFirstTimeThisTurn:boolean )
-  return false;
-end
-
 function OnInputHandler( pInputStruct:table )
```
I assume you left this function in because it was modified by CQUI, but you should remove it and just make a note in your PR mentioning that this functionality is now missing. I looked through the rest of the diff and don't really see where this would go now, but @Azurency might know.

This is just my opinion though, might want to wait with this until @Azurency comments.

If you do remove this function, make sure you remove the reference to it on [line 1213](https://github.com/Azurency/CQUI_Community-Edition/pull/384/files#diff-80f9bce53b0ae8a96e3056ffed4fc5c6R1213) as well (can't  comment on that line since it wasn't modified, I think).

tyrken replied:
Put KeyHandler back in so now I'm syntactically correct, will wait for further guidance.

ihendriks commented on Assets/UI/worldviewiconsmanager.lua:
```diff
@@ -338,7 +341,8 @@ function GetNonEmptyAt(plotIndex, state)
       if pPlot:IsStartingPlot() and WorldBuilder.IsActive() then
         pInstance = GetInstanceAt(plotIndex);
-        pInstance.RecommendationIconTexture:SetTexture( IconManager:FindIconAtlas("ICON_UNITOPERATION_FOUND_CITY", 43) );
+        -- CQUI updated 38->43 but 1.0.0.290 set it to 256
```
I'd put this comment in the PR, not in the code. But that's my opinion, don't know what @Azurency thinks.

tyrken replied:
Thanks - in affect you've now done this, will let him say which form he wants in future...

tyrken commented:
OK, for worldviewiconsmanager.lua I've added in the =>256 on line 93 and checked the feedback from #383.

TBH the second parameter to `IconManager:FindIconAtlas` is on closer inspection a size, so the new 256 value here & where I add the debatable comment just above might actually be wrong (UI from the mod might not have room compared to the new stock UI).  Should be pretty clear once we get the whole thing play-testable.

Azurency commented:
I'll try to make a review this week 😉

tyrken commented:
In unitflagmanager.lua do we need something to balance the CQUI instance.FlagBaseDarken on lines 522-536 (War Check), e.g. `instance.FlagBaseLighten:SetColor( primaryColor );`?

tyrken commented:
Got this comment in Screens/governmentscreen.xml: `<!--Filters are added dynamically through AddFilterTabButtons-->`
... hoping they actually meant RealizeFilterTabs in the matching lua file, added an equivalent there.

In line 1114 of Screens/governmentscreen.lua, I reverted a CQUI change of the variable name from "strSlotType" to "rowSlotType", that looked like a mistake.

tyrken commented:
* screens/civicstree.lua: Line 585 has CQUI removing an else that was there in original & update, I have NOT removed it as it looked wrong, please confirm!
```
if node["unlockGOV"] == nil then
  node["unlockGOV"] = InstanceManager:new( "GovernmentIcon", "GovernmentInstanceGrid", node.UnlockStack );
else   <--- This else was deleted by CQUI!
  node["unlockGOV"]:DestroyInstances()
end
```

* replacements/* - can you explain what is meant to be merged with what?  Confusing...

* popups/techciviccompletedpopup.lua - I can't really see where to put any of the changes from CQUI, so reverted to updated 290 version instead.
* popups/mappinlistpanel.xml - no changes needed

tyrken commented:
@Azurency have you managed to start code review yet?

Azurency commented:
@tyrken not this pull request, I'm on #382 for now

Azurency commented:
Great, thx for the contribution 👍 🎉
